src/main/java/seedu/address/model/person/Attendance.java

```diff
@@ -1,59 +1,67 @@
+ * The `Attendance` class represents a student's attendance, attendance status
+ * and participation record for a 12-week period.
  * It provides methods for managing and querying attendance information.
  */
 public class Attendance {
     public static final String TUTORIAL_ERROR_MSG = "Tutorial number is out of range, should be integer between 1-12";
     public static final String PARTICIPATION_ERROR_MSG = "PP number is out of range, should be >= 0";
-    public static final String ORIGINAL_ATD = "0,0,0,0,0,0,0,0,0,0,0,0";
+    public static final String STATUS_ERROR_MSG = "Status is invalid! It should either be P, VR or A.";
+    public static final String ORIGINAL_ATD = "U,U,U,U,U,U,U,U,U,U,U,U";
     public static final String ORIGINAL_PART = "0,0,0,0,0,0,0,0,0,0,0,0";
-    private int totalTut;
-    private final boolean[] attendanceList = new boolean[12];
-    private final int[] participationList = new int[12];
+    private static int totalTut = 12;
+    private final int[] participationList = new int[totalTut];
+    private final AttendanceStatus[] attendanceList = new AttendanceStatus[totalTut];
 
 
     /**
      * Constructs an `Attendance` object from a comma-separated attendance string.
      *
-     * @param atd A comma-separated string representing attendance for 1-12 weeks (e.g., "0,1,0,1,0,1,0,1,0,1,0,1").
+     * @param atd A comma-separated string representing attendance for 12 weeks (e.g., "0,1,0,2,0,3,0,1,0,1,0,1").
+     * @param pp A comma-separated string representing participation for 12 weeks (e.g., "120,0,100,130,110,150,100,100,
+     *           0,150,150,120").
      */
     public Attendance(String atd, String pp) {
         String[] atdArr = atd.split(",");
         String[] ppArr = pp.split(",");
-        this.totalTut = atdArr.length;
         for (int i = 0; i < totalTut; i++) {
-            if (atdArr[i].equals("1")) {
-                this.attendanceList[i] = true;
-                this.participationList[i] = Integer.parseInt(ppArr[i].trim());
-            } else {
-                this.participationList[i] = 0;
+            attendanceList[i] = new AttendanceStatus(i);
+            participationList[i] = Integer.parseInt(ppArr[i]);
+            switch (atdArr[i]) {
+            case "A":
+                attendanceList[i].setAbsent();
+                break;
+            case "P":
+                attendanceList[i].setPresent();
+                break;
+            case "VR":
+                attendanceList[i].setVR();
+                break;
+            default:
             }
         }
     }
 
     /**
-     * Checks if a given week is a valid tutorial number (between 1 and 12).
-     *
+     * Checks if a given week is a valid tutorial number (between 1 and the maximum tutorial count).
      * @param tutorial A string representing a tutorial number.
      * @return `true` if the week is valid; otherwise, `false`.
      */
     public static boolean isValidTutorial(String tutorial) {
```

Comment:
Not sure why you changed the method to a try catch when we are not throwing any errors. Does this count as abusing try catch? The previous method works fine.